File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    points.sort((a, b) => a[1] - b[1])
    let poppedUpTo = -Infinity, res = 0
    for (const [widthStart, widthEnd] of points) {
        if (widthStart > poppedUpTo) {
            poppedUpTo = widthEnd
            res++
        }
    }
    return res
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])
        popped_up_to, res = float("-inf"), 0
        for w0, w1 in points:
            if w0 > popped_up_to:
                popped_up_to = w1
                res += 1
        return res